Userland/Libraries/LibC/unistd.cpp

```diff
@@ -975,8 +975,49 @@ int unveil(char const* path, char const* permissions)
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/getpass.html
 char* getpass(char const* prompt)
 {
-    dbgln("FIXME: getpass('{}')", prompt);
-    TODO();
+    int tty = open("/dev/tty", O_RDWR | O_NOCTTY | O_CLOEXEC);
+    if (tty < 0)
+        return nullptr;
+
+    struct termios backup{};
+    if (tcgetattr(tty, &backup) < 0)
+        return nullptr;
```

Comment:
here, the open file descriptor for the tty is leaked. (and any early returns after it before closing the tty). Another scope guard.